src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java

```diff
@@ -297,26 +297,53 @@ private ClosePaymentRequestV2Dto.OutcomeEnum authorizationResultToOutcomeV2(
+    private Mono<TransactionRefundRequestedEvent> sendRefundRequestEvent(
+                                                                         Either<TransactionClosureErrorEvent, TransactionEvent<TransactionClosureData>> closureOutcomeEvent,
+                                                                         AuthorizationResultDto authorizationResult
     ) {
-        return Mono.just(closureEvent)
-                .filter(e ->
-                // Closed event sent on the queue only if the transaction was previously
-                // authorized and the Nodo response outcome is KO
-                TransactionClosureData.Outcome.KO.equals(e.getData().getResponseOutcome())
-                        && AuthorizationResultDto.OK.equals(authorizationResult)
+        return Mono.just(closureOutcomeEvent)
+                .filter(
+                        e -> e.fold(
+                                closureErrorEvent -> true,
+                                // Closed event sent on the queue only if the transaction was previously
+                                // authorized and the Nodo response outcome is KO
+                                closureEvent -> TransactionClosureData.Outcome.KO
+                                        .equals(closureEvent.getData().getResponseOutcome())
+                                        && AuthorizationResultDto.OK.equals(authorizationResult)
+                        )
                 )
-                .flatMap(e -> {
-                    log.info("Enqueue transaction closed event: {}", e);
-                    return transactionActivatedQueueAsyncClient
+                .map(
+                        e -> e.fold(
+                                closureErrorEvent -> {
+                                    log.info(
+                                            "Requesting refund for transaction {} because of bad or no response from Nodo",
+                                            closureErrorEvent.getTransactionId()
+                                    );
+                                    return closureErrorEvent.getTransactionId();
+                                },
+
+                                closureEvent -> {
+                                    log.info(
+                                            "Requesting refund for transaction {} as it was previously authorized but we either received KO response from Nodo",
+                                            closureEvent.getTransactionId()
+                                    );
+                                    return closureEvent.getTransactionId();
+                                }
+                        )
+                )
+                .flatMap(transactionId -> {
+                    TransactionRefundRequestedEvent refundRequestedEvent = new TransactionRefundRequestedEvent(
+                            transactionId,
+                            new TransactionRefundedData(TransactionStatusDto.CLOSED)
```

Comment:
here the previous state can be both `CLOSURE_ERROR` or `CLOSED`  depending on the TaskEither being left or right, right?

Reply:
You're right, fixing it

Reply:
Fixed in a20f88f